Python/bytecodes.c

```diff
@@ -3616,13 +3614,11 @@ dummy_func(
             PyObject *self = args[0];
             DEOPT_IF(!Py_IS_TYPE(self, method->d_common.d_type));
             DEOPT_IF(meth->ml_flags != METH_NOARGS);
+            // CPython promises to check all non-vectorcall function calls.
+            DEOPT_IF(tstate->c_recursion_remaining <= 0);
```

Comment:
The idea is that after DEOPT we will end up doing GOTO_ERROR in the original bytecode? (Is that what the comment is saying?)

Reply:
The comment refers to our obligation to check the C recursion limit for almost all builtin function calls.

We can fulfil that obligation more cheaply this way.